commands.c

```diff
@@ -1034,67 +1039,70 @@ int mutt_save_message(struct Mailbox *m, struct EmailList *el,
     app = en->email->security;
   }
   mutt_message_hook(m, en->email, MUTT_MESSAGE_HOOK);
-  mutt_default_save(buf, sizeof(buf), en->email);
+  mutt_default_save(buf->data, buf->dsize, en->email);
+  mutt_buffer_fix_dptr(buf);
 
-  mutt_pretty_mailbox(buf, sizeof(buf));
-  if (mutt_enter_fname(prompt, buf, sizeof(buf), false) == -1)
-    return -1;
+  mutt_buffer_pretty_mailbox(buf);
+  if (mutt_enter_fname(prompt, buf->data, buf->dsize, 0) == -1)
+    goto cleanup;
+  mutt_buffer_fix_dptr(buf);
 
-  size_t pathlen = strlen(buf);
+  size_t pathlen = mutt_buffer_len(buf);
   if (pathlen == 0)
-    return -1;
+    goto cleanup;
 
   /* Trim any trailing '/' */
-  if (buf[pathlen - 1] == '/')
-    buf[pathlen - 1] = '\0';
+  if (buf->data[pathlen - 1] == '/')
+    buf->data[pathlen - 1] = '\0';
 
   /* This is an undocumented feature of ELM pointed out to me by Felix von
```

Comment:
`unimportant`: You wouldn't happen to know what undocumented feature is mentioned here, would you?

Reply:
No, I wasn't aware of it.
Reading the code, it seems that saving a message to `.` means the last place you saved to, rather than the current directory of NeoMutt.

This doesn't seem to be documented in neomutt's manual either.
I wonder if we could kill this unexpected behaviour.

Reply:
> This doesn't seem to be documented in neomutt's manual either.
I wonder if we could kill this unexpected behaviour.

It might be safer to document the behavior. It's plausible that someone relies upon it.